indico/modules/events/registration/controllers/display.py

```diff
@@ -309,16 +309,16 @@ def _can_register(self):
             return False
         return True
 
-    def _process(self):
-        form = make_registration_form(self.regform)()
-        if self._can_register() and form.validate_on_submit():
-            registration = create_registration(self.regform, form.data, self.invitation)
-            return redirect(url_for('.display_regform', registration.locator.registrant))
-        elif form.is_submitted():
-            # not very pretty but usually this never happens thanks to client-side validation
-            for error in form.error_list:
-                flash(error, 'error')
+    def _process_POST(self):
+        schema = make_registration_schema(self.regform)()
+        form = parser.parse(schema)
+
+        if self._can_register():
+            registration = create_registration(self.regform, form, self.invitation)
+            return jsonify({'redirect': url_for('.display_regform', registration.locator.registrant)})
+        return self._process_GET()
```

Comment:
Not sure if this still makes so much sense - I'd just raise an `ExpectedError` if the user can't register since this is handled somewhat nicely by our JS (passed as a form-level error to final-form). I'd also do the check before even trying to parse the submitted data.